modules/core/src/main/java/org/apache/ignite/internal/client/thin/ReliableChannel.java

```diff
@@ -62,16 +68,19 @@
     private final Function<ClientChannelConfiguration, ClientChannel> chFactory;
 
     /** Client channel holders for each configured address. */
-    private final ClientChannelHolder[] channels;
+    private final AtomicReference<List<ClientChannelHolder>> channels = new AtomicReference<>();
 
     /** Index of the current channel. */
-    private int curChIdx;
+    private volatile int curChIdx = -1;
 
     /** Partition awareness enabled. */
     private final boolean partitionAwarenessEnabled;
 
     /** Cache partition awareness context. */
-    private final ClientCacheAffinityContext affinityCtx;
+    private final ClientCacheAffinityContext affCtx;
```

Comment:
Please revert unnecessary renames to reduce the scope of changes.

Reply:
Fixed.